exonum-java-binding/core/src/main/java/com/exonum/binding/util/LibraryLoader.java

```diff
@@ -46,26 +46,46 @@ private static void loadOnce() {
     try {
       System.loadLibrary(BINDING_LIB_NAME);
     } catch (UnsatisfiedLinkError e) {
-      logger.error("Failed to load '{}' library: {}…\n{}",
-          BINDING_LIB_NAME, e, extraLibLoadErrorInfo());
-      throw e;
+      String message = String.format("Failed to load '%s' library: %s.\n%s", BINDING_LIB_NAME,
+          e.getMessage(), extraLibLoadErrorInfo());
+      logger.fatal(message, e);
+
+      // Throw a new exception with a _full_ error message so that it is always available,
+      // even if the logger is not configured.
+      throw new LinkageError(message, e);
     }
   }
 
   private static String extraLibLoadErrorInfo() {
     String javaLibPath = System.getProperty(JAVA_LIBRARY_PATH_PROPERTY);
-    String dynamicLibVar = dynamicLibrariesEnvVar();
-    String dynamicLibPath = System.getenv(dynamicLibVar);
-    // todo: clarify this message when LD_LIBRARY_PATH becomes required.
-    return "java.library.path=" + javaLibPath + ", \n"
-        + dynamicLibVar + "=" + dynamicLibPath
-        + "\nMake sure that:\n"
-        + "1. The path to a directory containing '" + BINDING_LIB_NAME
-        + "' dynamic library image is included in either java.library.path system property or "
-        + dynamicLibVar + " environment variable.\n"
-        + "2. The paths to directories containing dynamic libraries required by '"
-        + BINDING_LIB_NAME + "', if any, are included in " + dynamicLibVar
-        + " environment variable";
+    if (runningUnitTests()) {
+      // 1. These error messages are tailored for an installed application. If you *develop*
+      // Exonum, the instructions might not be applicable.
+      // 2. java.library.path has a default platform-specific value if not specified (on 11),
+      // hence it might never be empty.
+      return String.format(
+          "Check that %s system property includes a path to '${EXONUM_HOME}/lib/native' directory\n"
+              + "containing %s library, where 'EXONUM_HOME' denotes the Exonum installation "
+              + "directory.\n"
+              + "The code launching tests must usually set it explicitly, see "
+              + "https://exonum.com/doc/version/0.10/get-started/java-binding/#testing",
```

Comment:
The description is a little confusing. The link doesn't say anything about Exonum installation. So it is not clear to me what path should I put for `EXONUM_HOME`. 
Also, I installed Exonum using https://exonum.com/doc/version/0.10/get-started/install and I do not see `/lib/native` directories inside neither the Exonum parent or `target` dirs.

Reply:
There is no installation section at the moment: https://jira.bf.local/browse/ECR-2990 and https://jira.bf.local/browse/ECR-3048 

ECR-2990 will update the testing section with appropriate surefire/failsafe configuration and link installation instructions.

Reply:
At the moment we can clarify "Exonum _Java_ installation directory", I think

Reply:
What about "Exonum Java *app* installation directory"?